gqcpy/src/QCMethod/HF/UHF/UHFSCFEnvironment_bindings.cpp

```diff
@@ -204,14 +199,22 @@ void bindUHFSCFEnvironment(py::module& module) {
 
         .def("replace_current_error_vectors_alpha",
              [](UHFSCFEnvironment<double>& environment, const Eigen::MatrixXd& new_error_vectors_alpha) {
-                 environment.error_vectors_alpha.pop_back();
-                 environment.error_vectors_alpha.push_back(SquareMatrix<double>(new_error_vectors_alpha));
+                 const auto last_spinResolved_error = environment.error_vectors.back();
+                 const auto last_error_vector_beta = last_spinResolved_error.beta();
+                 SpinResolved<VectorX<double>> new_error {new_error_vectors_alpha, last_error_vector_beta};
+
+                 environment.error_vectors.pop_back();
+                 environment.error_vectors.push_back(new_error);
              })
 
         .def("replace_current_error_vectors_beta",
              [](UHFSCFEnvironment<double>& environment, const Eigen::MatrixXd& new_error_vectors_beta) {
-                 environment.error_vectors_beta.pop_back();
-                 environment.error_vectors_beta.push_back(SquareMatrix<double>(new_error_vectors_beta));
+                 const auto last_spinResolved_error = environment.error_vectors.back();
+                 const auto last_error_vector_alpha = last_spinResolved_error.alpha();
+                 SpinResolved<VectorX<double>> new_error {last_error_vector_alpha, new_error_vectors_beta};
+
+                 environment.error_vectors.pop_back();
+                 environment.error_vectors.push_back(new_error);
              });
```

Comment:
In line of the spin-resolved types: I would suggest to change this into only providing `.def("replace_current_error_vectors")`. I think it's OK to ask the user to replace either (or both) of the `SpinResolved<...>` components and then push these changes to the environment.